code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/ConsentExtension.java

```diff
@@ -251,36 +219,46 @@ void handleEdgeConsentPreferenceHandle(final Event event) {
 	}
 
 	/**
-	 * Handles the get consents request event and dispatches a response event of EventType {@link ConsentConstants.EventType#CONSENT} and EventSource
-	 * {@link ConsentConstants.EventSource#RESPONSE_CONTENT} with the current consent details.
+	 * Handles the get consents request event and dispatches a response event of EventType {@link EventType#CONSENT} and EventSource
+	 * {@link EventSource#RESPONSE_CONTENT} with the current consent details.
 	 * <p>
 	 * Dispatched event will contain empty XDMConsentMap if currentConsents are null/empty.
 	 *
 	 * @param event the {@link Event} requesting consents
 	 */
 	void handleRequestContent(final Event event) {
-		ExtensionErrorCallback<ExtensionError> errorCallback = new ExtensionErrorCallback<ExtensionError>() {
-			@Override
-			public void error(final ExtensionError extensionError) {
-				MobileCore.log(
-					LoggingMode.DEBUG,
-					ConsentConstants.LOG_TAG,
-					String.format(
-						"ConsentExtension - Failed to dispatch %s event: Error : %s.",
-						ConsentConstants.EventNames.GET_CONSENTS_RESPONSE,
-						extensionError.getErrorName()
-					)
-				);
-			}
-		};
+		if (event == null) {
+			Log.trace(
+				ConsentConstants.LOG_TAG,
+				CLASS_NAME,
+				"%s - Unable to handle consent request as event is null.",
+				CLASS_NAME
+			);
+			return;
+		}
+
+		final Map<String, Object> eventData = event.getEventData();
+
+		if (isNullOrEmpty(eventData)) {
+			Log.trace(
+				ConsentConstants.LOG_TAG,
+				CLASS_NAME,
+				"%s - Unable to handle consent request event with id '%s': event data is missing or empty.",
+				CLASS_NAME,
+				event.getUniqueIdentifier()
+			);
+			return;
+		}
+
 		final Event responseEvent = new Event.Builder(
 			ConsentConstants.EventNames.GET_CONSENTS_RESPONSE,
-			ConsentConstants.EventType.CONSENT,
-			ConsentConstants.EventSource.RESPONSE_CONTENT
+			EventType.CONSENT,
+			EventSource.RESPONSE_CONTENT
```

Comment:
Should this event also include `.inResponseToEvent(event)` since this is a response event?